core/src/main/java/hudson/lifecycle/WindowsServiceLifecycle.java

```diff
@@ -120,6 +120,11 @@ public void rewriteHudsonWar(File by) throws IOException {
 
     @Override
     public void restart() throws IOException, InterruptedException {
+        Jenkins jenkins = Jenkins.getInstance();
+        if (jenkins != null) {
+            jenkins.cleanUp();
```

Comment:
Hi Daniel.

Thanks for your comment. I copy and pasted and same code that is on UnixLifecycle and SolarisSMFLifecycle, where there is also no validation.

The cleanUp method doesn't throw any checked exception, so I don't really know what type of exceptions to catch. Of course I can catch "Exception", but that it is not a good practice. Do you have any idea on what to catch?

This would also be a problem for other platforms, like I mentioned. I would probably suggest having another pull request to fix that situation for all platforms, but I am not really sure.

Thank you.